src/AbstractApplication.php

```diff
@@ -131,13 +158,54 @@ public function get($key, $default = null)
 	 */
 	public function getLogger()
 	{
-		// If a logger hasn't been set, use NullLogger
-		if (! ($this->logger instanceof LoggerInterface))
+		return $this->getContainer()->get('Psr\\Log\\LoggerInterface');
+	}
+
+	/**
+	 * Get the config of the application.
+	 *
+	 * @return  Joomla\Registry\Registry
+	 *
+	 * @since   2.0
+	 */
+	public function getConfig()
+	{
+		return $this->getContainer()->get('Joomla\\Registry\\Registry');
```

Comment:
If you HAVE to reference the config from the DI (I'm not totally sure what the benefit is yet), call `get('config')`.  The way you've done it now basically prevents you from using the DIC to build models implementing our ModelInterface (the model state is a Registry instance) or any of the packages which accept a Registry instance as their options store (which is quite a few packages).